packages/SystemsThinkingWorkbench-Core.package/STWConnection.class/instance/endNode..st

```diff
@@ -1,4 +1,7 @@
 accessing
 endNode: anObject
+	"Sets the end node of this connection and updates the respective nodes dependents accordingly."
 
-	endNode := anObject.
\ No newline at end of file
+	self endNode ifNotNilDo: [ :me | self endNode removeDependent: self].
```

Comment:
The block given to ifNotNilDo: will be called with one argument, the object which is not nil.

A block without a parameter will lead to an argument error.